src/hotspot/share/code/nmethod.hpp

```diff
@@ -479,21 +479,72 @@ class nmethod : public CodeBlob {
     not_used,
     OSR_invalidation_for_compiling_with_C1,
     OSR_invalidation_back_branch,
-    JVMCI_reprofile,
-    JVMCI_materialize_virtual_object,
-    JVMCI_invalidate_nmethod_mirror,
-    JVMCI_register_method,
     OSR_invalidation_of_lower_level,
     set_native_function,
     whitebox_deoptimization,
     missing_exception_handler,
     uncommon_trap,
     zombie,
+    JVMCI_reprofile,
+    JVMCI_materialize_virtual_object,
+    JVMCI_invalidate_nmethod_mirror,
+    JVMCI_register_method,
     JVMCI_invalidate_nmethod,
     JVMCI_new_installation,
     JVMCI_replacing_with_new_code,
   };
 
+
+  static const char* NMethodChangeReason_to_string(NMethodChangeReason reason) {
```

Comment:
Uh, use a switch:

```
 switch(reason) {
    case C1_deoptimize: return "C1 deoptimized";
    case C1_codepatch:  return "C1 code patch";
    ...
    default:
       assert(false, "Unhandled reason");
       return "Unknown";
 }
```

Reply:
Also, names: `change_reason_to_string(ChangeReason reason)`. Now that enum is scoped to `nmethod`, there is no need for `NMethod` prefix.